--- openjdk/hotspot/src/share/vm/runtime/arguments.cpp~	2013-12-27 22:22:03.820640995 +0100
+++ openjdk/hotspot/src/share/vm/runtime/arguments.cpp	2013-12-27 22:50:31.609181366 +0100
@@ -54,7 +54,7 @@
 #endif
 
 // Note: This is a special bug reporting site for the JVM
-#define DEFAULT_VENDOR_URL_BUG "http://icedtea.classpath.org/bugzilla"
+#define DEFAULT_VENDOR_URL_BUG "https://bugs.launchpad.net/ubuntu/+source/openjdk-6/"
 #define DEFAULT_JAVA_LAUNCHER  "generic"
 
 char**  Arguments::_jvm_flags_array             = NULL;
